File: src/Client/inventory-track/src/app/models/dto/writeoff-request/writeoff-request-response-dto.ts
import { DocumentInfoResponseDto } from "../document/document-info-response-dto";
import { WriteOffReasonResponseDto } from "../writeoff-reason/writeoff-reason-response-dto";
import { RequestStatusResponseDto } from "./request-status-response-dto";

export interface WriteOffRequestResponseDto {
  id: string;
  itemId: string;
  warehouseId: string;
  quantity: number;
  companyId: string;
  status: RequestStatusResponseDto;
  requestDate: string;
  approvedByUserId?: string;
  reason: WriteOffReasonResponseDto;
  documentId: string;
  requestedByUserId?: string; // 🆕 Кто создал запрос
  batchNumber?: string; // 🆕 Номер партии
}

File: src/Services/WriteOffService/WriteOffService.Application/DTOs/Response/WriteOffRequest/WriteOffRequestResponseDto.cs
using System;
using System.Collections.Generic;
using WriteOffService.Application.DTOs.Base;
using WriteOffService.Application.DTOs.Response.Document;
using WriteOffService.Application.DTOs.Response.RequestStatus;
using WriteOffService.Application.DTOs.Response.WriteOffReason;

namespace WriteOffService.Application.DTOs.Response.WriteOffRequest;

public class WriteOffRequestResponseDto : BaseDto
{
    public Guid ItemId { get; set; }
    public Guid WarehouseId { get; set; }
    public long Quantity { get; set; }
    public Guid CompanyId { get; set; }
    public RequestStatusResponseDto Status { get; set; }
    public DateTime RequestDate { get; set; }
    public Guid? ApprovedByUserId { get; set; }
    public WriteOffReasonResponseDto Reason { get; set; }
    public Guid DocumentId { get; set; }
}